Network cleanup

AnishReddyRavula opened the pull request:
NetworkMonitor - Check and Tear down improper blazar networks in neutron

- Iterate through each blazar network to find the networks that are not in 'active' or 'pending' reservation (maybe they are stuck in a lease)
- Tear down such networks in neutron which are not supposed to be in a user's project.
  1. Detach the network's subnets from the router.
  2. Delete the subnet(s)
  3. Delete the network

AnishReddyRavula committed:
returns the recent network's reservation

which is not in pending status and start_date of the reservation is less than current date

AnishReddyRavula committed:
Montor plugin for Network resources

With poll_resources() to teardown neutron networks while blazar's reservation/lease is in error/deleted state

AnishReddyRavula committed:
unittests for network plugin poll_resources

AnishReddyRavula committed:
Include monitor plugin in Network plugin

Mark-Powers commented on blazar/plugins/networks/network_plugin.py:
```diff
@@ -789,3 +793,117 @@ def _filter_networks_by_properties(self, network_properties,
+                # This means that network works fine as the reservation started fine
+                LOG.debug(f"Network {network_id_from_blazar} - VLAN {segment_id} is in active reservation {network_curr_reservation['id']} - skipping")
+                return
+            LOG.info(
```
Is this log statement accurate? This code seems to be no reservation because of the return above. I do not think we want a log statement here, as this would print something every time the monitor runs for every network. Additionally, I think this may raise an error if `network_curr_reservation` is `None`.

AnishReddyRavula replied:
This is supposed to be a debug statement and yes this line breaks when it is None

Mark-Powers reviewed (approved):
Code looks great! Seems very solid. Only a few minor comments about logging statements.

AnishReddyRavula committed:
Remove the log info message that

might break the method when network_curr_reservation is None.

The message serves not much purpose and may also flood the logs

Change-Id: If0f953256f049b6980ebfeeb9585761e3222e47c
